### PR TITLE
Fix latest profile

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Update profile builder `latest` command to respect the preview
subdirectory when omitting preview content.
Switched `latest` profile to use the `latest` command instead of a fixed
list that was outdated.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] The PR targets the `latest` branch.
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/1066

File: tools/profileBuilder/model/latest.go [hunk 1]
```diff
@@ -31,11 +31,16 @@ func acceptAll(name string) bool {
...
+func ignorePreview(name string) bool {
+	// check if the path contains a /preview/ subdirectory
+	if strings.Contains(name, previewSubdir) {
+		return false
+	}
 	matches := packageName.FindStringSubmatch(name)
 	version := matches[3]
-	result = !strings.Contains(version, "-preview") && !strings.Contains(version, "-beta") // matches[2] is the `version` group
-	return
+	return !strings.Contains(version, "-preview") && !strings.Contains(version, "-beta") // matches[2] is the `version` group
```
Comment (vladbarosan) [inline tools/profileBuilder/model/latest.go]: ok just to be sure, the policy is. if folder includes preview OR if its a preview api version we wont included. can you check please that the readme specifies this also for the latest profile

File: tools/profileBuilder/model/latest.go [hunk 2]
```diff
@@ -31,11 +31,16 @@ func acceptAll(name string) bool {
 	return true
 }
 
-func ignorePreview(name string) (result bool) {
+var previewSubdir = fmt.Sprintf("%spreview%s", string(os.PathSeparator), string(os.PathSeparator))
+
+func ignorePreview(name string) bool {
```
Comment (vladbarosan) [inline tools/profileBuilder/model/latest.go]: hmm not sure if there is a better name, but ignorePreview doesnt really tell me if it should return true or false if it is preview ( maybe `isPreview` )?

File: tools/profileBuilder/model/latest.go (hunk 1 above)
Comment (jhendrixMSFT) [reply tools/profileBuilder/model/latest.go]: The readme doesn't really explain the difference between the `latest` and `preview` profiles, will update.

**Comment by jhendrixMSFT** [comment]
I can confirm the content is deleted as there was a few preview APIs versions under latest that are now removed in this PR (they were in the list by mistake).

File: tools/profileBuilder/model/latest.go (hunk 2 above)
Comment (jhendrixMSFT) [reply tools/profileBuilder/model/latest.go]: Agreed the naming is confusing, will add a comment a clean up the names.

**Pull request merged**